tests/test_bot.py

```diff
@@ -0,0 +1,49 @@
+import time
+import pytest
+
+
+TIMEOUT = 60
+
+
+class TestAdminGroup:
+    def create_admingroup(self, admbot, botuser, db):
+        admchat = admbot.create_group_chat("admins", [], verified=True)
+        with db.write_transaction() as conn:
+            conn.set_config("admingrpid", admchat.id)
+        qr = admchat.get_join_qr()
+        chat = botuser.qr_join_chat(qr)
+        while not chat.is_protected():
+            time.sleep(0.1)
+        return chat
+
+    def test_help(self, admbot, botuser, db):
+        admingroup = self.create_admingroup(admbot, botuser, db)
+        admingroup.send_text("/help")
+        begin = time.time()
+        while len(admingroup.get_messages()) < 7:
+            time.sleep(0.1)
+            if time.time() > begin + TIMEOUT:
+                pytest.fail("Bot reply took more than %s seconds" % (TIMEOUT,))
+        assert admingroup.get_messages()[6].text.startswith("/add-user addr password token")
```

Comment:
the pytest-timeout plugin provides this generically for each test.  https://pypi.org/project/pytest-timeout/